libcudacxx/include/cuda/__ptx/instructions/generated/elect_sync.h

```diff
@@ -0,0 +1,36 @@
+// This file was automatically generated. Do not edit.
+
+#ifndef _CUDA_PTX_GENERATED_ELECT_SYNC_H_
+#define _CUDA_PTX_GENERATED_ELECT_SYNC_H_
+
+/*
+// elect.sync _|is_elected, membermask; // PTX ISA 80, SM_90
+template <typename = void>
+__device__ static inline bool elect_sync(
+  const uint32_t& membermask);
+*/
+#if __cccl_ptx_isa >= 800
+extern "C" _CCCL_DEVICE void __cuda_ptx_elect_sync_is_not_supported_before_SM_90__();
+template <typename = void>
+_CCCL_DEVICE static inline bool elect_sync(const _CUDA_VSTD::uint32_t& __membermask)
+{
+#  if _CCCL_CUDA_COMPILER(NVHPC) || __CUDA_ARCH__ >= 900
+  _CUDA_VSTD::uint32_t __is_elected;
+  asm volatile(
+    "{\n\t .reg .pred P_OUT; \n\t"
+    "elect.sync _|P_OUT, %1;\n\t"
+    "selp.b32 %0, 1, 0, P_OUT; \n"
```

Comment:
I am way out of my league here but I am curious why the generated code uses the `P_OUT` intermediate predicate to propagate the output to `%0` resulting having two commands (`elect.sync`, `selp`) and doesn't just do  `elect.sync %0|_, %1;`

Reply:
The reason is that inline PTX does not support predicate input or output arguments. So instead, we convert back and forth between a 32bit uint (bool).